src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala

```diff
@@ -56,6 +59,28 @@ case object ThurloeDatabaseConnector extends DataAccess with LazyLogging {
         Future.fromTry(databaseValuesToUserKeyValuePair(id, userId, key, value, iv))
     }
 
+  private def lookupSamUser(userId: String)(implicit samDAO: SamDAO): Future[sam.model.User] = {
+    val results = samDAO.getUserById(userId)
+
+    if (results.isEmpty) {
+      Future.failed(new KeyNotFoundException(userId, "n/a"))
+    } else if (results.size == 1) {
+      Future.successful(results.head)
+    } else {
+      // TODO once we have the newly generated models update this to use them
+      results
+        .find(x => Option(x.getAzureB2CId).isDefined)
+        .map(Future.successful)
+        .getOrElse(
+          Future.failed(
+            new InvalidDatabaseStateException(
+              s"Too many results returned from sam, none of which have a b2c id: ${results.size}"
```

Comment:
Based on our discussion, we should add the sam user ids into this log message so we can debug better if we run into this case